pytorch_lightning/core/hooks.py

```diff
@@ -115,6 +115,42 @@ def on_epoch_end(self) -> None:
         """
         # do something when the epoch ends
 
+    def on_train_epoch_start(self) -> None:
+        """
+        Called in the training loop at the very beginning of the epoch.
+        """
+        # do something when the epoch starts
+
+    def on_train_epoch_end(self) -> None:
+        """
+        Called in the training loop at the very end of the epoch.
+        """
+        # do something when the epoch ends
+
+    def on_val_epoch_start(self) -> None:
+        """
+        Called in the training loop at the very beginning of the epoch.
```

Comment:
```suggestion
        Called in the validation loop at the very beginning of the epoch.
```
same for the other hooks below